expan/core/results.py

```diff
@@ -0,0 +1,117 @@
+from expan.core.util import JsonSerializable
+
+
+# --------- Below are the data structure of statistics --------- #
+class BaseTestStatistics(JsonSerializable):
+    """ Holds only statistics for the control and treatment group. 
+    
+    :param control_statistics: statistics within the control group
+    :type  control_statistics: SampleStatistics
+    :param treatment_statistics: statistics within the treatment group
+    :type  treatment_statistics: SampleStatistics
+    """
+    def __init__(self, control_statistics, treatment_statistics):
+        self.control_statistics   = control_statistics
+        self.treatment_statistics = treatment_statistics
+
+
+class SampleStatistics(JsonSerializable):
+    """ This class holds sample size, mean and variance.
+    
+    :type sample_size: int
+    :type mean: float
+    :type variance: float
+    """
+    def __init__(self, sample_size, mean, variance):
+        self.sample_size = sample_size
+        self.mean        = mean
+        self.variance    = variance
+
+
+class SimpleTestStatistics(BaseTestStatistics):
+    """ Additionally to BaseTestStatistics, holds delta, confidence interval, statistical power, and p value.
+    
+    :type control_statistics: SampleStatistics
+    :type treatment_statistics: SampleStatistics
+    :type delta: float
+    :type p: float
+    :type statistical_power: float
+    :param ci: a dict where keys are percentiles and values are the corresponding value for the statistic.
+    :type  ci: dict
+    """
+    def __init__(self, control_statistics, treatment_statistics, delta, ci, p, statistical_power):
+        super(SimpleTestStatistics, self).__init__(control_statistics, treatment_statistics)
+        self.delta               = delta
+        self.p                   = p
+        self.statistical_power   = statistical_power
+        # TODO: think of structure {p: v} (the same as ci)
+        self.confidence_interval = [{'percentile': p, 'value': v} for (p, v) in ci.items()]
```

Comment:
Didn't we have a class for `ConfidenceInterval`?

Reply:
We had a class previously, but then we found that it leads to two times of the fields:
```
"confidence_interval": {
    "confidence_interval" : [
        {"percentile": 2.5, "value": -1},
        {"percentile": 97.5.5, "value": 1}
    ]
}

Reply:
The ```TODO``` here is to think of whether we want:
```
    "confidence_interval" : [
        {"percentile": 2.5, "value": -1},
        {"percentile": 97.5.5, "value": 1}
    ]
```
or
```
    "confidence_interval" : [
        {"2.5": -1},
        {"97.5.5", 1}
    ]
```

Reply:
I see, thank you for clarifying this one for me.

Reply:
the problem I see here with structure {'2.5': 0.5, '97.5': 0.4} is that everywhere where we need to access those fields we always need to know those numbers (2.5 and 97.5). If they are fixed, then it's ok-ish, but things can change if we decide to have another percentiles (for example) or add some intermediate (we allowed that before as far as I remember).

Reply:
you are right. Then I think we can stay with ``` {"percentile": 2.5, "value": -1}```.